core/repository/api/src/main/java/org/eclipse/rdf4j/repository/RepositoryConnection.java

```diff
@@ -622,7 +623,39 @@ public void exportStatements(Resource subj, IRI pred, Value obj, boolean include
 	 * @see #rollback()
 	 * @see #setIsolationLevel(IsolationLevel)
 	 */
-	public void begin(IsolationLevel level) throws RepositoryException;
+	void begin(IsolationLevel level) throws RepositoryException;
+
+	/**
+	 * Begins a new transaction with the supplied {@link TransactionSetting}, requiring {@link #commit()} or
+	 * {@link #rollback()} to be called to end the transaction.
+	 *
+	 * @param settings The {@link TransactionSetting} (zero or more) for this transaction. If an isolation level is
+	 *                 provided in the settings this will be used for the transaction. If none is provided then the
+	 *                 default will be used. Behaviour of this method is undefined if more than one isolation level is
+	 *                 provided. Behaviour of this method is undefined if one or more settings is null.
+	 * @throws RepositoryException If the connection could not start the transaction. Possible reasons this may happen
+	 *                             are:
+	 *                             <ul>
+	 *                             <li>a transaction is already {@link #isActive() active} on the current connection.
+	 *                             <li>the specified {@link IsolationLevel} is not supported by the store, and no
+	 *                             compatible level could be found.
+	 *                             </ul>
+	 * @see #begin()
+	 * @see #isActive()
+	 * @see #commit()
+	 * @see #rollback()
+	 * @see #setIsolationLevel(IsolationLevel)
+	 */
+	default void begin(TransactionSetting... settings) {
```

Comment:
Added java docs now.

Reply:
@jeenbroekstra what do you think?

Reply:
Looks good to me! One last niggle: add a `@since 3.3.0` if you have time.

Reply:
LGTM, and I agree that `@since` tags are very useful on new public API methods.